openquake/hazardlib/source/point.py

```diff
@@ -171,6 +175,18 @@ def iter_ruptures(self, **kwargs):
                         surface, occurrence_rate,
                         self.temporal_occurrence_model)
 
+    def point_ruptures(self):
+        """
+        Generate one point rupture for each magnitude
+        """
+        _, np = self.nodal_plane_distribution.data[0]
```

Comment:
This is the only thing I do not like. I think we should take the hypocenter and nodal plane with the highest weight.

Reply:
At long distances it should not matter much, but it is easy to change.